cos-fleet-catalog-connectors-it-support/src/main/java/org/bf2/cos/connector/camel/it/support/ConnectorContainer.java

```diff
@@ -177,6 +179,21 @@ protected void containerIsStarting(InspectContainerResponse containerInfo) {
         }
     }
 
+    @Override
+    protected void containerIsStopping(InspectContainerResponse containerInfo) {
+        super.containerIsStopped(containerInfo);
+        int timeoutSeconds = 30;
+        LOGGER.info("Container is stopping. Attempting to wait for graceful stop for {} seconds.", timeoutSeconds);
+        try (StopContainerCmd stopContainerCmd = getDockerClient()
+                .stopContainerCmd(getContainerId())
+                .withTimeout(timeoutSeconds)) {
+            stopContainerCmd.exec();
+        } catch (Exception e) {
+            LOGGER.error("Failed to gracefully stop container, this might lead to resource leaking.", e);
+        }
+        LOGGER.info("Container was gracefully stopped.");
```

Comment:
this may be moved inside the try block otherwise you may get a warning about container not gracefully stopped and then a message about the container being gracefully stopped

Reply:
ooops. fixed.